src/ScreenWidget.py

```diff
@@ -1,160 +1,94 @@
 #!/usr/bin/python
-
-from PyQt5.QtWidgets import QWidget, QLabel, QVBoxLayout, QHBoxLayout, QPushButton, QMenu
-from PyQt5.QtGui import QPainter, QPixmap, QImage, QColor, qRgb, qRgba
-from PyQt5.QtCore import Qt, pyqtSignal, QRect
-
-import numpy as np
+import functools
 import cv2
+from PyQt5.QtWidgets import QWidget, QVBoxLayout, QAction
+from PyQt5.QtCore import pyqtSignal
+from PyQt5 import QtCore
+import pyqtgraph as pg
 
 from Helper import *
 
-WIDTH_SCREEN = WS =  500
-HEIGHT_SCREEN = HS = 375
-
-CONVERSION_PX = WIDTH_FRAME / WIDTH_SCREEN
-
 
-class ScreenWidget(QLabel):
+class ScreenWidget(pg.GraphicsView):
 
     selected = pyqtSignal(int, int)
     cleared = pyqtSignal()
 
     def __init__(self, filename=None, model=None, parent=None):
-        QWidget.__init__(self, parent=parent)
+        super().__init__(parent=parent)
         self.filename = filename
         self.model = model
 
-        self.zoom = False
+        self.viewBox = pg.ViewBox()
+        self.setCentralItem(self.viewBox)
+        self.viewBox.setAspectLocked()
+        self.viewBox.invertY()
 
-        self.setMinimumSize(WIDTH_SCREEN, HEIGHT_SCREEN)
-        self.setMaximumSize(WIDTH_SCREEN, HEIGHT_SCREEN)
+        self.imageItem = ClickableImage()
+        self.viewBox.addItem(self.imageItem)
+        self.imageItem.mouseClicked.connect(self.imageClicked)
 
-        self.pixmap = QPixmap(WIDTH_SCREEN, HEIGHT_SCREEN)
-        self.painter = QPainter(self.pixmap)
+        self.clickTarget = pg.TargetItem()
+        self.viewBox.addItem(self.clickTarget)
+        self.clickTarget.setVisible(False)
 
-        self.frame = QImage(WIDTH_FRAME, HEIGHT_FRAME, QImage.Format_RGB32)
-        self.overlay = QImage(WIDTH_FRAME, HEIGHT_FRAME, QImage.Format_ARGB32)
-        self.overlay.fill(qRgba(0,0,0,0))
+        self.cameraActions = []
+        self.cameraActionSeparator = self.viewBox.menu.insertSeparator(self.viewBox.menu.actions()[0])
 
         if self.filename:
             self.setData(cv2.imread(filename, cv2.IMREAD_GRAYSCALE))
 
-        self.display()
-
         self.clearSelected()
 
         self.camera = None
 
-    def zoomIn(self):
-        self.zoom = True
-
-    def zoomOut(self):
```

Comment:
we need this for the calibration procedure; this function is called anytime a target point is reached by the probe to ensure the full FOV is visible